scripts/ci/build.sh

```diff
@@ -9,10 +9,13 @@ if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" -o -n "$FORCE_RELEASE" ]
 then
     echo -n "" >prerelease.txt
 else
-    # Use last commit date rather than build date to avoid ending up with builds for
-    # different platforms having different version strings (and therefore producing different bytecode)
-    # if the CI is triggered just before midnight.
-    TZ=UTC git show --quiet --date="format-local:%Y.%-m.%-d" --format="ci.%cd" >prerelease.txt
+    # Checks whether the prerelease file does not exists or if it does, it is not empty and is not a nightly build
+    if [ ! -e ./prerelease.txt ] || { [ -s ./prerelease.txt ] && [ "$(< ./prerelease.txt cut -d "." -f 1)" != "nightly" ]; }; then
```

Comment:
This really feels like a hack. It's is completely invisible to the caller and just silently modifies things. How about adding an optional parameter to pass in the identifier instead?

Or, alternatively, introducing something like `nightly.txt` that makes the script ignore the existing `prerelease.txt`?

Either of those would be more explicit.

Reply:
hahaha ;)
But it could be done in a function instead if that helps. The point was that I didn't want to change the current way builds versions are made (based on the `prerelease.txt` file), since I don't know all the parts that rely on it.

But it does not modify anything new. I mean, before, it was setting the `prerelease.txt` as a `ci` version for all cases, which was wrong for `nightly` builds. Now, it just skips if it is a nightly build.

Sure, I can make it a parameter. But it will also require changes in the solc-bin as well.

Reply:
No problem with changing solc-bin. Just submit a PR there.

And this will not be changing anything about the `prerelease.txt` mechanism. Just changing how the file is generated so no worries. Let's add the parameter.